File: 2016-04-20-AndroidTestExample/src/main/java/tech/thdev/androidtestexample/base/BaseActivity.java
package tech.thdev.androidtestexample.base;

import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import butterknife.ButterKnife;

/**
 * Created by Tae-hwan on 4/20/16.
 */
public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getContentViewId());

        ButterKnife.bind(this);

        onCreate();
    }

    @LayoutRes
    protected abstract int getContentViewId();

    protected abstract void onCreate();
}

File: webviewexample/src/main/kotlin/tech/thdev/webviewjavascriptinterface/base/view/BaseActivity.kt
package tech.thdev.webviewjavascriptinterface.base.view

import android.support.v7.app.AppCompatActivity
import butterknife.ButterKnife

/**
 * Created by Tae-hwan on 8/8/16.
 */
abstract class BaseActivity : AppCompatActivity() {

    override fun setContentView(layoutResID: Int) {
        super.setContentView(layoutResID)
        ButterKnife.bind(this@BaseActivity)
    }
}